models/dataset.go

```diff
@@ -457,21 +457,54 @@ func ValidateDataset(ctx context.Context, dataset *Dataset) error {
 	var invalidFields []string
 	if dataset.URI != "" {
 		dataset.URI = strings.TrimSpace(dataset.URI)
-		datasetURI, err := url.Parse(dataset.URI)
+		_, err := url.Parse(dataset.URI)
 		if err != nil {
 			invalidFields = append(invalidFields, "URI")
 			log.Event(ctx, "error parsing URI", log.ERROR, log.Error(err))
-		} else {
-			if !strings.Contains(datasetURI.Path, "/datasets") || !strings.Contains(datasetURI.Path, dataset.ID) {
-				fmt.Println("hello")
-				invalidFields = append(invalidFields, "URI")
+		}
+	}
+
+	for i := range dataset.Publications {
+		generalDetails := &dataset.Publications[i]
+		if generalDetails != nil {
+			generalDetails.HRef = strings.TrimSpace(generalDetails.HRef)
+			_, err := url.Parse(generalDetails.HRef)
+			if err != nil {
+				invalidFields = append(invalidFields, "href")
+				log.Event(ctx, "error parsing URI", log.ERROR, log.Error(err))
+			}
+		}
+	}
+
+	for i := range dataset.Methodologies {
+		generalDetails := &dataset.Methodologies[i]
```

Comment:
As discussed each use of this logic is operating on a `GeneralDetails` type so we can pull this out into a func that accepts a list of `GeneralDetails`  instead of duplicating the logic.